api/client_rest_test.go

```diff
@@ -0,0 +1,23 @@
+package api
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestRESTGetDelete(t *testing.T) {
+	http := &FakeHTTP{}
+
+	client := NewClient(
+		ReplaceTripper(http),
+		AddHeader("Authorization", "Basic BASE64ENCODE_CLIENT_ID:CLIENT_SECRET"),
+		AddHeader("Content-Type", "text/plain"),
+	)
+
+	http.StubResponse(204, bytes.NewBuffer([]byte{}))
+
+	r := bytes.NewReader([]byte(`{"access_token": "ACCESS_TOKEN"}`))
```

Comment:
Since the request body won't be used for anything, you could pass an empty JSON body like so `{}` instead of setting `access_token` which is not relevant for this test

Reply:
Agreed